hudi-common/src/main/java/org/apache/hudi/common/table/log/BaseHoodieLogRecordReader.java

```diff
@@ -410,6 +418,34 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
     }
   }
 
+  private static void prepareRecordPositions(String blockPositionStr,
+                                             List<Integer> blockPositions,
+                                             Set<Integer> globalPositions,
+                                             boolean positionBasedMergeEnabled,
+                                             boolean useEventOrdering) throws IOException {
+    if (!positionBasedMergeEnabled || blockPositionStr == null || blockPositionStr.isEmpty()) {
+      return;
+    }
+
+    Roaring64NavigableMap posBitmap = PositionUtils.deserializeCodecStringWithRoaring64NavigableMap(blockPositionStr);
```

Comment:
Can we not directly use `HoodieLogBlock#getRecordPositions`? Something like below:
```
Roaring64NavigableMap recordPositions = logBlock.getRecordPositions();
if (recordPositions != null && !recordPositions.isEmpty()) {
    recordPositions.iterator();
    ...
}
```

Reply:
Right!